torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -257,3 +263,54 @@ def lexicon_decoder(
         sil_token=sil_token,
         unk_word=unk_word,
     )
+
+
+def _get_filenames(model: str) -> _PretrainedFiles:
+    MODEL_FILE_MAP = {
+        "librispeech": _PretrainedFiles(lexicon="lexicon.txt", tokens="tokens.txt", lm=None),
+        "librispeech-3-gram": _PretrainedFiles(lexicon="lexicon.txt", tokens="tokens.txt", lm="lm.bin"),
+        "librispeech-4-gram": _PretrainedFiles(lexicon="lexicon.txt", tokens="tokens.txt", lm="lm.bin"),
+    }
+
+    files = MODEL_FILE_MAP[model]
+    prefix = f"decoder-assets/{model}"
+    return _PretrainedFiles(
+        lexicon=f"{prefix}/{files.lexicon}",
+        tokens=f"{prefix}/{files.tokens}",
+        lm=f"{prefix}/{files.lm}" if files.lm is not None else None,
+    )
+
+
+def download_pretrained_files(model: str) -> _PretrainedFiles:
+    """
+    Retrieves pretrained data files used for CTC decoder.
+
+    Args:
+        model (str): pretrained language model to download
+            Options: ["librispeech-3-gram", "librispeech-4-gram", "librispeech"]
+
+    Returns:
+        Object with the following attributes:
+            lm: path corresponding to downloaded language model, or None if lm not provided
+            lexicon: path corresponding to downloaded lexicon file
+            tokens: path corresponding to downloaded tokens file
+    """
+
+    hub_dir = torch.hub.get_dir()
+    download_path = f"{hub_dir}/{model}"
+    if not os.path.isdir(download_path):
+        os.makedirs(download_path)
```

Comment:
L299-L302 are now handled by `download_asset` function, so we can get rid of it.